test/L2GovToken.t.sol

```diff
@@ -30,6 +34,84 @@ contract L2GovTest is L2GovTestPreInit {
     tokenProxy.grantRole(tokenProxy.BURNER_ROLE(), burner);
     vm.stopPrank();
   }
+
+  function _createValidPartialDelegation(uint256 _n, uint256 _seed) internal view returns (PartialDelegation[] memory) {
+    _seed = bound(
+      _seed,
+      1,
+      /* private key can't be bigger than secp256k1 curve order */
+      115_792_089_237_316_195_423_570_985_008_687_907_852_837_564_279_074_904_382_605_163_141_518_161_494_337 - 1
+    );
+    _n = _n != 0 ? _n : (_seed % tokenProxy.MAX_PARTIAL_DELEGATIONS()) + 1;
+    PartialDelegation[] memory delegations = new PartialDelegation[](_n);
+    uint96 _totalNumerator;
+    for (uint256 i = 0; i < _n; i++) {
+      uint96 _numerator = uint96(
+        bound(
+          uint256(keccak256(abi.encode(_seed + i))) % tokenProxy.DENOMINATOR(), // initial value of the numerator
+          1,
+          tokenProxy.DENOMINATOR() - _totalNumerator - (_n - i) // ensure that there is enough numerator left for the
+            // remaining delegations
+        )
+      );
+      delegations[i] = PartialDelegation(address(uint160(uint160(vm.addr(_seed)) + i)), _numerator);
+      _totalNumerator += _numerator;
+    }
+    return delegations;
+  }
+
+  function _expectEmitDelegateVotesChangedEvents(
```

Comment:
thanks @garyghayrat , is it possible to move these helpers into a helper file that both the L2GovToken tests and ERC20VotesPartialDelegation tests can inherit from?

Reply:
Added it here: https://github.com/voteagora/ERC20VotesPartialDelegationUpgradeable/pull/43/commits/df32083843be552a2d3de7e2d9e2c31fb2cac87a